### ISSUE TITLE
Tooltip hidden when inside a container with overflow: auto

### ISSUE THREAD

**Issue by transitorytammy** [sir-dunxalot/ember-tooltips#279]
For embertooltips 3.0, since the tooltip is now a sibling to the element, there is an issue when a parent container has `overflow: auto/scroll`, the tooltip sometimes gets cut off.

<img width="385" alt="screen shot 2018-09-28 at 2 24 51 pm" src="https://user-images.githubusercontent.com/1967604/46234316-bbb45300-c32a-11e8-9ec1-ca1db230d90d.png">

Is there an option to show the tooltip in this scenario?

**Comment by maxfierke** [comment]
@transitorytammy you may want to experiment with passing a `targetId` to `{{ember-tooltip}}`, which will allow you to target a different parent element

Let me know if that works for your use case, as this came up with the travis-ci folks recently too, so it might be good for us to document in the coming upgrade documentation!

**Comment by transitorytammy** [comment]
hey @maxfierke!  I did try doing that but it doesn't work in my case.  It might be because there are a few levels of divs between the parent with the `overflow-y: auto` and where the tooltip is.

**Comment by maxfierke** [comment]
Interesting... would you be able to put together a small example that reproduces your use case's DOM structure? (doesn't need to be a live example, just a snippet that I could use to write a test or build out a new ember app with to play around)

**Comment by transitorytammy** [comment]
So, the DOM structure is basically like this.

```
<div class="outerContainer">
  <table>
    {{#each sortedTerms as |keyword index|}}
      <tr class="row">
        <td class="term">
          {{#ember-tooltip}}
            {{keyword.term.hint}}
          {{/ember-tooltip}}
          <span>{{keyword.text}}</span>
        </td>
        <td class="strength">
          {{#ember-tooltip}}
            {{keyword.term.hint}}
          {{/ember-tooltip}}
          <span>{{keyword.text}}</span>
        </td>
      </tr>
  </table>
</div>
```

and `.outerContainer` has `overflow-y: auto` on it that is causing the issues for me.

**Comment by maxfierke** [comment]
@transitorytammy I played around with this a bit. It seems to work as expected if `position` is unset or `static`, but if `.outerContainer` is `position: relative` or `position: absolute` it exhibits this behavior. Is `.outerContainer` or another parent of the tooltip positioned this way by any chance? 

If so, I was able to resolve that behavior by removing the `preventOverflow` option we pass into popper.js (added in 3.0.0-beta.3). We're hard-coding the popper options right now

@sir-dunxalot: it seems like we may want to make that configurable. Given the breadth of options available in popper.js, what do you think about exposing a `popperOptions` property that would allow users to pass in additional options (and merge with the defaults we pass in, if needed.)

(Probably also the problem in #275, as well)

**Comment by sir-dunxalot** [comment]
Adding a `popperOptions` property seems like a natural progression towards customization without introducing much complexity. So I'm for that addition. We should consider whether passing `popperOptions` will overwrite the whole default object or merge the passed object and our default object. I lean towards the latter since our defaults are simple and easy to overwrite:

https://github.com/sir-dunxalot/ember-tooltips/blob/3.x/addon/components/ember-tooltip-base.js#L310

**Comment by maxfierke** [comment]
@transitorytammy 3.0.0-beta.5 adds support for passing `popperOptions`. I think something like this should fix the behavior you're seeing:

`{{ember-tooltips text=tooltipText popperOptions=popperOptions}}`

where `popperOptions` is set to something like this:

```json
{
  "modifiers": {
    "preventOverflow": {
      "escapeWithReference": false
    }
  }
}
```

**Comment by sir-dunxalot** [comment]
Given @maxfierke's workaround, this is fixed by #272.

**Comment by transitorytammy** [comment]
Awesome.  Thank you @maxfierke and @sir-dunxalot  for this update.  It works like a charm now.  :)
